salt/payload.py

```diff
@@ -130,14 +150,30 @@ def load(self, fn_):
         data = fn_.read()
         fn_.close()
         if data:
-            return self.loads(data)
+            if six.PY3:
+                return self.loads(data, encoding='utf-8')
+            else:
+                return self.loads(data)
 
-    def dumps(self, msg):
+    def dumps(self, msg, use_bin_type=False):
         '''
         Run the correct dumps serialization format
+
+        :param use_bin_type: Useful for Python 3 support. Tells msgpack to
+                             differentiate between 'str' and 'bytes' types
+                             by encoding them differently.
+                             Since this changes the wire protocol, this
+                             option should not be used outside of IPC.
         '''
         try:
-            return msgpack.dumps(msg)
+            if msgpack.version >= (0, 4, 0):
```

Comment:
I didn't check but am I right to assume that `msgpack.version` is a tuple of int's?